config/routes.rb

```diff
@@ -1,3 +1,7 @@
 Rails.application.routes.draw do
   # For details on the DSL available within this file, see http://guides.rubyonrails.org/routing.html
+  get '/guests/:id', to: 'guests#show'
+  post '/guests/:id/guest_rooms', to: 'guest_rooms#create'
+
+  get '/rooms', to: 'rooms#index'
 end
```

Comment:
The controller and action are perfect. You want the route, however, to be representative of the association we are creating if we are abiding by REST principles.

Even though we ARE creating a new row in the guest_rooms table, we want the route to look something like,  `post "/guests/:id/rooms"`. This says that we are creating a new room linked to the guest with that particular ID.

If you wanted to use `guest_rooms` in your route, then something like `post "/guest_rooms"` would be better, the only thing here is that you would need to pass the guest ID in a hidden field in your form.

---

There may be other opinions when it comes to this, but if we think about how the route would look if we are using nested resources with the `resources` method. It's parent on the outside and child on the inside. So:

```ruby
resources :guests, only: :show do
  resources :rooms, only: :create
end
```

This nested resource would produce 2 paths: `get "/guests/:id"` and `post "/guests/:guest_id/rooms"`, so when we handroll our routes we want to stick to those conventions as well.